src/main/java/Duke.java

```diff
@@ -13,34 +59,21 @@ public static void main(String[] args) {
         System.out.println("What can I do for you?" + System.lineSeparator() + boundary);
         Scanner in = new Scanner(System.in);
         String line = in.nextLine();
-        Task[] taskList = new Task[100];
-        int countTask = 0;
+
 
         while (!line.equalsIgnoreCase("bye")) {
             if (line.equalsIgnoreCase("list")) {
-                System.out.print(boundary);
-                for (int i = 0; i < countTask; i ++) {
-                    System.out.println((i + 1) + ". [" + taskList[i].getStatusIcon() + "] " + taskList[i].description);
-                }
-                System.out.print(boundary);
+                printList();
             } else if (line.toLowerCase().startsWith("mark")) {
-                int toMark = Integer.parseInt(line.substring(5)) - 1;
-                taskList[toMark].markDone();
-                System.out.println(boundary + "Nice! I've marked this task as done:");
-                System.out.println("[" + taskList[toMark].getStatusIcon() + "] " + taskList[toMark].description + System.lineSeparator() + boundary);
+                markTask(Integer.parseInt(line.substring(5)) - 1);
             } else if (line.toLowerCase().startsWith("unmark")) {
-                int toMark = Integer.parseInt(line.substring(7)) - 1;
-                taskList[toMark].markNotDone();
-                System.out.println(boundary + "OK, I've marked this task as not done yet:");
-                System.out.println("[" + taskList[toMark].getStatusIcon() + "] " + taskList[toMark].description + System.lineSeparator() + boundary);
+                unmarkTask(Integer.parseInt(line.substring(7)) - 1);
             } else {
-                taskList[countTask] = new Task(line);
-                countTask ++;
-                System.out.println(boundary + "added: " + line + System.lineSeparator() + boundary);
+                addTask(line);
             }
             in = new Scanner(System.in);
             line = in.nextLine();
         }
-        System.out.print(boundary + "Bye. Hope to see you again soon!" + System.lineSeparator() + boundary);
+        sayGoodbye();
```

Comment:
I like how you've abstracted the printing of your exit message. But perhaps the same can also be done for your welcome message and execution of inputs?